src/ape/managers/chain.py

```diff
@@ -416,10 +416,27 @@ def revert_to_block(self, block_number: int):
             block_number (int): The block number to revert to.
         """
 
-        self._map = {
+        self._address_list_map = {
             a: [r for r in receipts if r.block_number <= block_number]
             for a, receipts in self.items()
         }
+        self._hash_to_receipt_map = {
+            h: r for h, r in self._hash_to_receipt_map.items() if r.block_number <= block_number
+        }
+
+    def get_receipt(self, transaction_hash: str) -> Optional[ReceiptAPI]:
+        """
+        Get a receipt from the history by its transaction hash.
+
+        Args:
+            transaction_hash (str): The hash of the transaction.
+
+        Returns:
+            Optional[:class:`~ape.api.transactions.ReceiptAPI`]: The receipt, if it
+              is present in the transaction history. Else, ``None``.
+        """
+
+        return self._hash_to_receipt_map.get(transaction_hash)
```

Comment:
Curious if this is better 
```suggestion
        return self._hash_to_receipt_map.get(transaction_hash) or self.provider.get_receipt(transaction_hash)
```